src/server/sample_data/routes/list.js

```diff
@@ -0,0 +1,35 @@
+import { createIndexName } from './lib/create_index_name';
+
+export const createListRoute = () => ({
+  path: '/api/sample_data',
+  method: 'GET',
+  config: {
+    handler: async (request, reply) => {
+      const { callWithRequest } = request.server.plugins.elasticsearch.getCluster('data');
+
+      const sampleDatasets = request.server.getSampleDatasets().map(sampleDataset => {
+        return {
+          id: sampleDataset.id,
+          name: sampleDataset.name,
+          description: sampleDataset.description,
+          previewImagePath: sampleDataset.previewImagePath,
+          overviewDashboard: sampleDataset.overviewDashboard,
+          defaultIndex: sampleDataset.defaultIndex,
+        };
+      });
+
+      const isInstalledPromises = sampleDatasets.map(async sampleDataset => {
+        const index = createIndexName(request.server, sampleDataset.id);
+        try {
+          const { count } = await callWithRequest(request, 'count', { index: index });
+          sampleDataset.isInstalled = count > 0;
+        } catch (err) {
+          sampleDataset.isInstalled = false;
```

Comment:
think this might also hide permission errors.

Reply:
Hmm, should we distinguish between permission error here and not installed?

Otherwise we will show 'install' to a user without read permissions on the data indices.  If we catch a permissions error here, maybe we just shouldn't show any action buttons, or disable them with a tooltip?

Reply:
I reworked the isInstalled check to return status instead of a boolean. That way, if there is an error during the checking process than the UI can just show a disabled add button and surface the error the the user